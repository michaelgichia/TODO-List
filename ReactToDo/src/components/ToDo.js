import React, { Component } from 'react';
import _ from 'lodash';
import TodoInput from './TodoInput';
import DisplayToDos from './DisplayToDos';

export default class ToDo extends Component {
	constructor(props){
    super(props)
    this.state = {
      todos: [
        {content: "Learn Redux in Depth", isCompleted: true},
        {content: "Learn Redux Saga in Depth", isCompleted: false},
        {content: "Learn Redux Redux-Form in Depth", isCompleted: false}
      ],
      term: ''
    }
    this.createTodo = this.createTodo.bind(this)
    this.handelSubmit = this.handelSubmit.bind(this)
    this.onInputChange = this.onInputChange.bind(this)
    this.renderTodos = this.renderTodos.bind(this)
    this.deleteTodo = this.deleteTodo.bind(this)
    this.updateTodos = this.updateTodos.bind(this)
  }

  onInputChange(event) {
    this.setState({term: event.target.value})
  }

  createTodo(value) {
    let todos = _.concat(this.state.todos, { content: value, isCompleted: false})
    this.setState({todos})
  }

  handelSubmit(event) {
    event.preventDefault();
    if(this.refs.inputValue.props.value.length < 5) {return}
    const value = this.refs.inputValue.props.value
    this.createTodo(value)
    this.setState({term: ""})
  }

  deleteTodo(done) {
    const {todos} = this.state
  	_.remove(todos, todo => todo === done )
    this.setState({todos})
  }

  updateTodos(newContent, oldContent) {
  	const {todos} = this.state
  	const oldTodoObj = _.find(todos, todo => todo.content === oldContent)
  	oldTodoObj.content = newContent
  	this.setState({todos})
  }

  renderTodos(todos) {
  	return(_.map(todos, (todo, index) => (
  		<DisplayToDos
				key={index}
				{...todo}
				{...this.updateTodos}
        id={`checkbox-${index}`}
        HtmlFor={`checkbox-${index}`}
				onDelete={() => this.deleteTodo(todo)}
				updateTodos={this.updateTodos}
			/>
		)))
  }

  render() {
    return(
      <div>
        <div className="title col l6 m8 s12 offset-l3 offset-m2">
          <p>To~Do</p>
        </div>
        <div className="col l6 m8 s12 offset-l3 offset-m2">
          <TodoInput
            createTodo={this.createTodo}
            onSubmit={this.handelSubmit}
            ref="inputValue"
            onChange={this.onInputChange}
            value={this.state.term}/>
          </div>
          <div className="col l6 m8 s12 offset-l3 offset-m2">
            <ul>
            	{this.renderTodos(this.state.todos)}
            </ul>
          </div>
      </div>
    )
  }
}
