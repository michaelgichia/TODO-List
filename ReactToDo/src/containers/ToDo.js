import React, {Component, PropTypes} from 'react';
import _ from 'lodash';
import TodoInput from './TodoInput';
import DisplayToDos from './DisplayToDos';
import Task from './Task';

export default class ToDo extends Component {
	constructor(props){
    super(props)
    this.state = {
      todos: [
        {content: "Learn Redux in Depth", isCompleted: true},
        {content: "Learn Redux Saga in Depth", isCompleted: false},
        {content: "Learn Redux Redux-Form in Depth", isCompleted: false}
      ],
      active: false,
      complete: false
    }
  }

  deleteTodo = (done) => {
    const {todos} = this.state
  	_.remove(todos, todo => todo === done )
    this.setState({todos})
  }

  updateTodos = (newContent, oldContent) => {
  	const {todos} = this.state
  	const oldTodoObj = _.find(todos, todo => todo.content === oldContent)
  	oldTodoObj.content = newContent
  	this.setState({todos})
  }

  updateCheckbox = (oldCheckboxVal) => {
    const {todos} = this.state
    const newValue = _.find(todos, todo => todo.content === oldCheckboxVal.content)
    newValue.isCompleted = !newValue.isCompleted;
    this.setState({todos});
  }

  updateState = (prevstate) => {
    this.setState({todos: prevstate})
  }

  displayActiveTodos = () => {
    this.setState(
      {active: true, complete: false}
    )
  }

  displayCompletedTodos = () => {
    this.setState({active: false, complete: true})
  }

  displayAllTodos = () => {
    this.setState({active: false, complete: false})
  }

  playActions = () => {
    const {todos, active, complete} = this.state;
    if(active) {
      return _.filter(todos, todo => todo.isCompleted === false);
    } else if(complete) {
      return _.filter(todos, todo => todo.isCompleted === true);
    } else if(!complete && !active) {
      return todos;
    } else {
      return todos;
    }
  }

  renderTodos = () => {
    let todoItems = this.playActions();
  	return(_.map(todoItems, (todo, index) => (
  		<DisplayToDos
				key={index}
        {...this.state}
				{...todo}
				{...this.updateTodos}
        id={index}
        HtmlFor={index}
				onDelete={() => this.deleteTodo(todo)}
				updateTodos={this.updateTodos}
        updateCheckbox={this.updateCheckbox}
			/>
		)))
  }

  render() {
    return(
      <div>
        <div className="title col l6 m8 s12 offset-l3 offset-m2">
          <p>To~Do</p>
        </div>
        <div className="col l6 m8 s12 offset-l3 offset-m2">
          <TodoInput />
          <Task
            {...this.state}
            updateState={this.updateState} 
            displayActiveTodos={this.displayActiveTodos}
            displayCompletedTodos={this.displayCompletedTodos}
            displayAllTodos={this.displayAllTodos}
          />
        </div>
          <div className="col l6 m8 s12 offset-l3 offset-m2">
            <ul>
              {this.renderTodos()}
            </ul>
          </div>
      </div>
    )
  }
}

ToDo.propTypes = {
  value: PropTypes.number,
};
