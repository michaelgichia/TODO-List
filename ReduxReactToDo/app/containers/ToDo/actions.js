import {
  ADD_TODO,
  TOGGLE_TODO,
  EDIT_TODO,
  DELETE_TODO
} from './constants';

export const addTodo = (input) => ({type: ADD_TODO, payload: input})
export const completeTodo = (id) => ({type: TOGGLE_TODO, id})
export const editTodo = (id, input) => ({type: EDIT_TODO, id, input})
export const deleteTodo = (id) => ({type: DELETE_TODO, id})

