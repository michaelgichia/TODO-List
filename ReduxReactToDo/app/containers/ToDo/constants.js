/*
 *
 * ToDo constants
 *
 */

export const DEFAULT_ACTION = 'app/ToDo/DEFAULT_ACTION';
