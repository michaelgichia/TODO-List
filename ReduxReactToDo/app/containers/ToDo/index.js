import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import makeSelectToDo from './selectors';
import NavBar from 'components/NavBar';
import Box from 'components/Box';
import TodoInputText from 'components/TodoInputText';
import TodoDisplay from 'components/TodoDisplay';

export class ToDo extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    return (
      <div>
        <header>
          <NavBar />
        </header>
        <Box>
          <TodoInputText />
          <TodoDisplay />
        </Box>
      </div>
    );
  }
}

ToDo.propTypes = {
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = createStructuredSelector({
  ToDo: makeSelectToDo(),
});

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(ToDo);
