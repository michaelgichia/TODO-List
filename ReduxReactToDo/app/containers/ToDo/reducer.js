import { fromJS } from 'immutable';
import _ from 'lodash';
import {
  ADD_TODO,
  TOGGLE_TODO,
  EDIT_TODO,
  DELETE_TODO
} from './constants';


const initialState = [
	{
		content: "es6, react, redux, sagas, material-ui, styled-components",
		isCompleted: true,
		id: 0
	}
];


function toDoReducer(state = initialState, action) {
  switch (action.type) {
    case ADD_TODO:
    	return [...state, 
    		{
          content: action.payload, 
    		  isCompleted: false, 
    		  id: state.reduce((id, todo) => Math.max(todo.id, id), -1) + 1
        }
      ]

    case TOGGLE_TODO:
      return state.map(todo =>
        todo.id === action.id ?
        { ...todo, isCompleted: !todo.isCompleted } : todo)

    case EDIT_TODO:
      return state.map(todo => todo.id === action.id ?
        {...todo, content: action.input} : todo
      )

    case DELETE_TODO:
      return state.filter(todo => todo.id !== action.id)

    default:
      return state;
  }
}

export default toDoReducer;
