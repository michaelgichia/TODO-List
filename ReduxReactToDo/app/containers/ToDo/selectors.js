import { createSelector } from 'reselect';

/**
 * Direct selector to the toDo state domain
 */
const selectToDoDomain = () => (state) => state.get("toDo");

/**
 * Other specific selectors
 */


/**
 * Default selector used by ToDo
 */

const makeSelectToDo = () => createSelector(
  selectToDoDomain(),
  (substate) => substate
);

export {
  selectToDoDomain,
};
export default makeSelectToDo;
