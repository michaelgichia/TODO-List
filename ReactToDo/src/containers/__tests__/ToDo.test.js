import React from 'react';
import { mount, shallow } from 'enzyme';

import ToDo from '../ToDo';

describe('ToDo components', () => {
  const props = {};
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<ToDo {...props}/>);
  });

  it('renders correctly', () => {
    const tree = shallow(<ToDo />);
    expect(tree).toMatchSnapshot();
  });
});
