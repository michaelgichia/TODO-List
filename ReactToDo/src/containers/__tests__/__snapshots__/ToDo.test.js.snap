// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo components renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="title col l6 m8 s12 offset-l3 offset-m2"
    >
        <p>
            To~Do
        </p>
    </div>
    <div
        className="col l6 m8 s12 offset-l3 offset-m2"
    >
        <TodoInput
            createTodo={[Function]}
        />
        <Task
            active={false}
            complete={false}
            displayActiveTodos={[Function]}
            displayAllTodos={[Function]}
            displayCompletedTodos={[Function]}
            todos={
                Array [
                    Object {
                      "content": "Learn Redux in Depth",
                      "isCompleted": true,
                    },
                    Object {
                      "content": "Learn Redux Saga in Depth",
                      "isCompleted": false,
                    },
                    Object {
                      "content": "Learn Redux Redux-Form in Depth",
                      "isCompleted": false,
                    },
                  ]
            }
            updateState={[Function]}
        />
    </div>
    <div
        className="col l6 m8 s12 offset-l3 offset-m2"
    >
        <ul>
            <DisplayToDos
                HtmlFor={0}
                active={false}
                complete={false}
                content="Learn Redux in Depth"
                id={0}
                isCompleted={true}
                onDelete={[Function]}
                todos={
                    Array [
                        Object {
                          "content": "Learn Redux in Depth",
                          "isCompleted": true,
                        },
                        Object {
                          "content": "Learn Redux Saga in Depth",
                          "isCompleted": false,
                        },
                        Object {
                          "content": "Learn Redux Redux-Form in Depth",
                          "isCompleted": false,
                        },
                      ]
                }
                updateCheckbox={[Function]}
                updateTodos={[Function]}
            />
            <DisplayToDos
                HtmlFor={1}
                active={false}
                complete={false}
                content="Learn Redux Saga in Depth"
                id={1}
                isCompleted={false}
                onDelete={[Function]}
                todos={
                    Array [
                        Object {
                          "content": "Learn Redux in Depth",
                          "isCompleted": true,
                        },
                        Object {
                          "content": "Learn Redux Saga in Depth",
                          "isCompleted": false,
                        },
                        Object {
                          "content": "Learn Redux Redux-Form in Depth",
                          "isCompleted": false,
                        },
                      ]
                }
                updateCheckbox={[Function]}
                updateTodos={[Function]}
            />
            <DisplayToDos
                HtmlFor={2}
                active={false}
                complete={false}
                content="Learn Redux Redux-Form in Depth"
                id={2}
                isCompleted={false}
                onDelete={[Function]}
                todos={
                    Array [
                        Object {
                          "content": "Learn Redux in Depth",
                          "isCompleted": true,
                        },
                        Object {
                          "content": "Learn Redux Saga in Depth",
                          "isCompleted": false,
                        },
                        Object {
                          "content": "Learn Redux Redux-Form in Depth",
                          "isCompleted": false,
                        },
                      ]
                }
                updateCheckbox={[Function]}
                updateTodos={[Function]}
            />
        </ul>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="title col l6 m8 s12 offset-l3 offset-m2"
      >
            <p>
                  To~Do
            </p>
      </div>
      <div
            className="col l6 m8 s12 offset-l3 offset-m2"
      >
            <TodoInput
                  createTodo={[Function]}
            />
            <Task
                  active={false}
                  complete={false}
                  displayActiveTodos={[Function]}
                  displayAllTodos={[Function]}
                  displayCompletedTodos={[Function]}
                  todos={
                        Array [
                              Object {
                                "content": "Learn Redux in Depth",
                                "isCompleted": true,
                              },
                              Object {
                                "content": "Learn Redux Saga in Depth",
                                "isCompleted": false,
                              },
                              Object {
                                "content": "Learn Redux Redux-Form in Depth",
                                "isCompleted": false,
                              },
                            ]
                  }
                  updateState={[Function]}
            />
      </div>
      <div
            className="col l6 m8 s12 offset-l3 offset-m2"
      >
            <ul>
                  <DisplayToDos
                        HtmlFor={0}
                        active={false}
                        complete={false}
                        content="Learn Redux in Depth"
                        id={0}
                        isCompleted={true}
                        onDelete={[Function]}
                        todos={
                              Array [
                                    Object {
                                      "content": "Learn Redux in Depth",
                                      "isCompleted": true,
                                    },
                                    Object {
                                      "content": "Learn Redux Saga in Depth",
                                      "isCompleted": false,
                                    },
                                    Object {
                                      "content": "Learn Redux Redux-Form in Depth",
                                      "isCompleted": false,
                                    },
                                  ]
                        }
                        updateCheckbox={[Function]}
                        updateTodos={[Function]}
                  />
                  <DisplayToDos
                        HtmlFor={1}
                        active={false}
                        complete={false}
                        content="Learn Redux Saga in Depth"
                        id={1}
                        isCompleted={false}
                        onDelete={[Function]}
                        todos={
                              Array [
                                    Object {
                                      "content": "Learn Redux in Depth",
                                      "isCompleted": true,
                                    },
                                    Object {
                                      "content": "Learn Redux Saga in Depth",
                                      "isCompleted": false,
                                    },
                                    Object {
                                      "content": "Learn Redux Redux-Form in Depth",
                                      "isCompleted": false,
                                    },
                                  ]
                        }
                        updateCheckbox={[Function]}
                        updateTodos={[Function]}
                  />
                  <DisplayToDos
                        HtmlFor={2}
                        active={false}
                        complete={false}
                        content="Learn Redux Redux-Form in Depth"
                        id={2}
                        isCompleted={false}
                        onDelete={[Function]}
                        todos={
                              Array [
                                    Object {
                                      "content": "Learn Redux in Depth",
                                      "isCompleted": true,
                                    },
                                    Object {
                                      "content": "Learn Redux Saga in Depth",
                                      "isCompleted": false,
                                    },
                                    Object {
                                      "content": "Learn Redux Redux-Form in Depth",
                                      "isCompleted": false,
                                    },
                                  ]
                        }
                        updateCheckbox={[Function]}
                        updateTodos={[Function]}
                  />
            </ul>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ToDo />,
      "_debugID": 47,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ToDo {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "createTodo": [Function],
        "deleteTodo": [Function],
        "displayActiveTodos": [Function],
        "displayAllTodos": [Function],
        "displayCompletedTodos": [Function],
        "playActions": [Function],
        "props": Object {},
        "refs": Object {},
        "renderTodos": [Function],
        "state": Object {
          "active": false,
          "complete": false,
          "todos": Array [
            Object {
              "content": "Learn Redux in Depth",
              "isCompleted": true,
            },
            Object {
              "content": "Learn Redux Saga in Depth",
              "isCompleted": false,
            },
            Object {
              "content": "Learn Redux Redux-Form in Depth",
              "isCompleted": false,
            },
          ],
        },
        "updateCheckbox": [Function],
        "updateState": [Function],
        "updateTodos": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 13,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="title col l6 m8 s12 offset-l3 offset-m2"
          >
                    <p>
                              To~Do
                    </p>
          </div>
          <div
                    className="col l6 m8 s12 offset-l3 offset-m2"
          >
                    <TodoInput
                              createTodo={[Function]}
                    />
                    <Task
                              active={false}
                              complete={false}
                              displayActiveTodos={[Function]}
                              displayAllTodos={[Function]}
                              displayCompletedTodos={[Function]}
                              todos={
                                        Array [
                                                  Object {
                                                    "content": "Learn Redux in Depth",
                                                    "isCompleted": true,
                                                  },
                                                  Object {
                                                    "content": "Learn Redux Saga in Depth",
                                                    "isCompleted": false,
                                                  },
                                                  Object {
                                                    "content": "Learn Redux Redux-Form in Depth",
                                                    "isCompleted": false,
                                                  },
                                                ]
                              }
                              updateState={[Function]}
                    />
          </div>
          <div
                    className="col l6 m8 s12 offset-l3 offset-m2"
          >
                    <ul>
                              <DisplayToDos
                                        HtmlFor={0}
                                        active={false}
                                        complete={false}
                                        content="Learn Redux in Depth"
                                        id={0}
                                        isCompleted={true}
                                        onDelete={[Function]}
                                        todos={
                                                  Array [
                                                            Object {
                                                              "content": "Learn Redux in Depth",
                                                              "isCompleted": true,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Saga in Depth",
                                                              "isCompleted": false,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Redux-Form in Depth",
                                                              "isCompleted": false,
                                                            },
                                                          ]
                                        }
                                        updateCheckbox={[Function]}
                                        updateTodos={[Function]}
                              />
                              <DisplayToDos
                                        HtmlFor={1}
                                        active={false}
                                        complete={false}
                                        content="Learn Redux Saga in Depth"
                                        id={1}
                                        isCompleted={false}
                                        onDelete={[Function]}
                                        todos={
                                                  Array [
                                                            Object {
                                                              "content": "Learn Redux in Depth",
                                                              "isCompleted": true,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Saga in Depth",
                                                              "isCompleted": false,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Redux-Form in Depth",
                                                              "isCompleted": false,
                                                            },
                                                          ]
                                        }
                                        updateCheckbox={[Function]}
                                        updateTodos={[Function]}
                              />
                              <DisplayToDos
                                        HtmlFor={2}
                                        active={false}
                                        complete={false}
                                        content="Learn Redux Redux-Form in Depth"
                                        id={2}
                                        isCompleted={false}
                                        onDelete={[Function]}
                                        todos={
                                                  Array [
                                                            Object {
                                                              "content": "Learn Redux in Depth",
                                                              "isCompleted": true,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Saga in Depth",
                                                              "isCompleted": false,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Redux-Form in Depth",
                                                              "isCompleted": false,
                                                            },
                                                          ]
                                        }
                                        updateCheckbox={[Function]}
                                        updateTodos={[Function]}
                              />
                    </ul>
          </div>
</div>,
        "_debugID": 48,
        "_renderedOutput": <div>
          <div
                    className="title col l6 m8 s12 offset-l3 offset-m2"
          >
                    <p>
                              To~Do
                    </p>
          </div>
          <div
                    className="col l6 m8 s12 offset-l3 offset-m2"
          >
                    <TodoInput
                              createTodo={[Function]}
                    />
                    <Task
                              active={false}
                              complete={false}
                              displayActiveTodos={[Function]}
                              displayAllTodos={[Function]}
                              displayCompletedTodos={[Function]}
                              todos={
                                        Array [
                                                  Object {
                                                    "content": "Learn Redux in Depth",
                                                    "isCompleted": true,
                                                  },
                                                  Object {
                                                    "content": "Learn Redux Saga in Depth",
                                                    "isCompleted": false,
                                                  },
                                                  Object {
                                                    "content": "Learn Redux Redux-Form in Depth",
                                                    "isCompleted": false,
                                                  },
                                                ]
                              }
                              updateState={[Function]}
                    />
          </div>
          <div
                    className="col l6 m8 s12 offset-l3 offset-m2"
          >
                    <ul>
                              <DisplayToDos
                                        HtmlFor={0}
                                        active={false}
                                        complete={false}
                                        content="Learn Redux in Depth"
                                        id={0}
                                        isCompleted={true}
                                        onDelete={[Function]}
                                        todos={
                                                  Array [
                                                            Object {
                                                              "content": "Learn Redux in Depth",
                                                              "isCompleted": true,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Saga in Depth",
                                                              "isCompleted": false,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Redux-Form in Depth",
                                                              "isCompleted": false,
                                                            },
                                                          ]
                                        }
                                        updateCheckbox={[Function]}
                                        updateTodos={[Function]}
                              />
                              <DisplayToDos
                                        HtmlFor={1}
                                        active={false}
                                        complete={false}
                                        content="Learn Redux Saga in Depth"
                                        id={1}
                                        isCompleted={false}
                                        onDelete={[Function]}
                                        todos={
                                                  Array [
                                                            Object {
                                                              "content": "Learn Redux in Depth",
                                                              "isCompleted": true,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Saga in Depth",
                                                              "isCompleted": false,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Redux-Form in Depth",
                                                              "isCompleted": false,
                                                            },
                                                          ]
                                        }
                                        updateCheckbox={[Function]}
                                        updateTodos={[Function]}
                              />
                              <DisplayToDos
                                        HtmlFor={2}
                                        active={false}
                                        complete={false}
                                        content="Learn Redux Redux-Form in Depth"
                                        id={2}
                                        isCompleted={false}
                                        onDelete={[Function]}
                                        todos={
                                                  Array [
                                                            Object {
                                                              "content": "Learn Redux in Depth",
                                                              "isCompleted": true,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Saga in Depth",
                                                              "isCompleted": false,
                                                            },
                                                            Object {
                                                              "content": "Learn Redux Redux-Form in Depth",
                                                              "isCompleted": false,
                                                            },
                                                          ]
                                        }
                                        updateCheckbox={[Function]}
                                        updateTodos={[Function]}
                              />
                    </ul>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ToDo />,
}
`;
